examples/CloudRetro/README.md

```diff
@@ -0,0 +1,509 @@
+
+```console
+kubectl apply -f - <<EOF
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: stunner
+  namespace: stunner
+spec:
+  selector:
+    matchLabels:
+      app: stunner
+  template:
+    metadata:
+      labels:
+        app: stunner
+    spec:
+      containers:
+        - command: ["stunnerd"]
+          args: ["-w", "-c", "/etc/stunnerd/stunnerd.conf"]
+          image: l7mp/stunnerd:latest
+          imagePullPolicy: Always
+          name: stunnerd
+          env:
+            - name: STUNNER_ADDR
+              valueFrom:
+                fieldRef:
+                  apiVersion: v1
+                  fieldPath: status.podIP
+          volumeMounts:
+            - name: stunnerd-config-volume
+              mountPath: /etc/stunnerd
+      volumes:
+        - name: stunnerd-config-volume
+          configMap:
+            name: stunnerd-configmap
+EOF
+```
+
+Next up, we will deploy an actual STUNner gateway.
+The below Gateway specification will expose the STUNner gateway over the STUN/TURN listener service running on the UDP listener port 3478. STUnner will await clients to connect to this listener port and, once authenticated, let them connect to the services running inside the Kubernetes cluster; meanwhile, the NAT traversal functionality implemented by the STUN/TURN server embedded into STUNner will make sure that clients can connect from behind even the most over-zealous enterprise NAT or firewall.
+
+```console
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: Gateway
+metadata:
+  name: udp-gateway
+  namespace: stunner
+spec:
+  gatewayClassName: stunner-gatewayclass
+  listeners:
+    - name: udp-listener
+      port: 3478
+      protocol: UDP
+EOF
+```
+
+This will create us a whole new service:
+
+```console
+kubectl get services -n stunner
+
+NAME                              TYPE           CLUSTER-IP      EXTERNAL-IP                  PORT(S)          AGE
+stunner                           LoadBalancer   10.85.128.58    <STUNNER_EXTERNAL_IP>        3478:31362/UDP   13m
+stunner-gateway-udp-gateway-svc   LoadBalancer   10.85.131.128   <GATEWAY_EXTERNAL_IP>        3478:32576/UDP   30s
+```
+
+
+Note down <GATEWAY_EXTERNAL_IP> and the port assigned to it (if not using NodePort, 3478 in our case). This will be the address the clients will need to connect to for the TURN server.
+
+
+Finally, attach a UDP route to the Gateway, so that clients will be able to connect via the public STUN/TURN listener UDP:3478 to the worker LoadBalancer service we`ve created.
+In our case, we named it worker-ci-udp-svc. Don`t forget to specify the namespace, even if its in the default one.
+
+```console
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: UDPRoute
+metadata:
+  name: worker-udp-route
+  namespace: stunner
+spec:
+  parentRefs:
+    - name: udp-gateway
+  rules:
+    - backendRefs:
+        - name: worker-ci-udp-svc
+          namespace: cloudretro
```

Comment:
This is the part that's interesting to the user! here is where we attach stunner to the cloudretro setup!